### PR TITLE
Explicit select SHA256 for code signing on windows

### PR DESCRIPTION
**Pull request by daschuer**
This fixes the following error message: 

```
SignTool Error: No file digest algorithm specified. Please specify the digest algorithm with the /fd flag. Using /fd SHA256 is recommended and more secure than SHA1. Calling signtool with /fd sha1 is equivalent to the previous behavior. In order to select the hash algorithm used in the signing certificate's signature, use the /fd certHash option. 
```
This restores the original behavior as a band aid. I have no clue what it means to switch to the recommended /fd SHA256

### CODE REVIEW

**Comment by Swiftb0y** [comment]
IMO its just a change in hash strength. Changing it to the more secure sha256 should make no difference IMO. Do we hardcode the hashes somewhere else?

**Comment by daschuer** [comment]
We can just give it a try. I will add a commit on top of this for SHA256 merge it and ask a windows user if anything it wrong. In case yes, quickly revert. Unfortunately we need to do this in main because if the secrets.

**Comment by Swiftb0y** [comment]
Does it really have to be on main? Can't we just quickly create a different branch based on main in the mixxx repo?

**Comment by Swiftb0y** [comment]
https://github.com/mixxxdj/mixxx/tree/signtool_sha256

File: CMakeLists.txt [hunk 1]
```diff
@@ -1326,9 +1326,9 @@ if(WIN32)
     set(WINDOWS_CODESIGN_CERTIFICATE_PATH "$ENV{WINDOWS_CODESIGN_CERTIFICATE_PATH}" CACHE STRING "Path to signtool certificate")
     set(WINDOWS_CODESIGN_CERTIFICATE_PASSWORD "$ENV{WINDOWS_CODESIGN_CERTIFICATE_PASSWORD}" CACHE STRING "Password of signtool certificate")
     if("${WINDOWS_CODESIGN_CERTIFICATE_PATH}" STREQUAL "" AND "${WINDOWS_CODESIGN_CERTIFICATE_PASSWORD}" STREQUAL "")
-        set(WINDOWS_CODESIGN_ARGS /a /t http://timestamp.verisign.com/scripts/timstamp.dll CACHE STRING "parameters for signtool (list)")
+        set(WINDOWS_CODESIGN_ARGS /fd SHA256 /a /t http://timestamp.verisign.com/scripts/timstamp.dll CACHE STRING "parameters for signtool (list)")
```
Comment (Swiftb0y) [inline CMakeLists.txt:1329]: we also need to specify the timestamp digest algorithm.
See the note at the top of the page: https://docs.microsoft.com/en-us/windows/win32/seccrypto/signtool

Comment (Swiftb0y) [reply CMakeLists.txt]: I'm actually not sure about that anymore. We don't actually timestamp the build right now. I don't know what the purpose of that would be.

**Comment by Swiftb0y** [comment]
You also need to set the timestamp digest algorithm
https://github.com/mixxxdj/mixxx/commit/d60036155301c24d68002f8bc3919bf589300a02
https://github.com/mixxxdj/mixxx/runs/7031560221

The build succeeded but it doesn't actually do any timestamping because we didn't specify `/t`.If we do specify the timestamp server, the build fails because the timestamp url is invalid?!?

**Comment by daschuer** [comment]
Closed in favor of https://github.com/mixxxdj/mixxx/pull/4824

**Pull request closed without merging**